flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -700,6 +700,13 @@ private static void setupAlarm(
       AlarmManager alarmManager,
       long epochMilli,
       PendingIntent pendingIntent) {
+
+    if (notificationDetails.scheduleMode == null) {
+      // Handle ScheduleMode.useAllowWhileIdle() called on a null object reference
+      // Issue reference: https://github.com/MaikuB/flutter_local_notifications/issues/2136
+      notificationDetails.scheduleMode = ScheduleMode.inexact;
```

Comment:
Thanks for the PR but this looks like it hasn't been updated that I mentioned in https://github.com/MaikuB/flutter_local_notifications/issues/2136#issuecomment-1806769446 where the wrong schedule mode has been picked. It should be exact to retain the original behaviour as per https://github.com/MaikuB/flutter_local_notifications/blob/4b723e750d1371206520b10a122a444c4bba7475/flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java#L549

Reply:
sorry, may be i didn't understand the requirement. do you want me to add this? 

`if (notificationDetails.scheduleMode == null) {
        checkCanScheduleExactAlarms(alarmManager);
        AlarmManagerCompat.setExact( alarmManager, AlarmManager.RTC_WAKEUP, epochMilli, pendingIntent);
    }`

Reply:
That's not what I mean though is close. Currently you have specified inexact timing to be used via

```
notificationDetails.scheduleMode = ScheduleMode.inexact;
```

When I said this is wrong and should be exact instead, this means it should fixed to be

```
notificationDetails.scheduleMode = ScheduleMode.exact
```

This should be done instead of what you wrote as there would be code below to make sure exact alarm permissions have been granted as that needs to be done before trying to set an exact alarm based on that code you asked about.

Another thing is could you also change the code comment as mentioning the GitHub issue link it fixes isn't the important part. More important to link to https://github.com/MaikuB/flutter_local_notifications/blob/master/flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java#L735 that has the code for the old behaviour. More specifically can you change it to be the following?

```
// This is to account for notifications created in older versions prior to allowWhileIdle
// being added so the deserialiser.
// Reference to old behaviour:
// https://github.com/MaikuB/flutter_local_notifications/blob/master/flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java#L735
```

This would make the comment consistent with another similar fix

Reply:
great. Thanks. Will do.

Reply:
Hi. I have implemented the changes you suggested. Kindly review.

Reply:
Left a comment elsewhere about how code isn't compiling